.github/workflows/e2e.yml

```diff
@@ -0,0 +1,66 @@
+---
+name: E2E-CI
+
+on:  # yamllint disable-line rule:truthy
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+jobs:
+  run-e2e:
+    runs-on: ubuntu-latest
+    environment:
+      name: dev
+    steps:
+      - name: Set environmental variables
+        run: |
+          echo "OC_PROJECT=test-e2e-$GITHUB_SHA" >> $GITHUB_ENV
+      - uses: actions/checkout@v1
+      - name: Prepare environment
+        run: |
+          # Install dependencies
+          pip install --user openshift
+
+          # Install tkn cli
+          curl -LO https://github.com/tektoncd/cli/releases/download/v0.11.0/tkn_0.11.0_Linux_x86_64.tar.gz
+          tar xvzf tkn_0.11.0_Linux_x86_64.tar.gz -C /usr/local/bin/
+
+          # Prepare Kubeconfig.
+          # Use GitHub secret which contains the Kubeconfig for SA with admin privilleges (same as in ansible vault)
+          # to create Kubernetes context.
+          mkdir $HOME/.kube
+          cat <<EOF > $HOME/.kube/config
+          ${{ secrets.KUBECONFIG }}
+          EOF
+
+          # Prepare vault password
+          echo ${{ secrets.VAULT_PASSWORD }} > ansible/vault-password
```

Comment:
This seems insecure to me on the off chance these commands end up in logs. What can be done to ensure these won't appear in GitHub at all?

Reply:
Correct me if I'm wrong, but I'm not seeing any vulnerability here. If it would appear in the logs, it should be redacted by GitHub. 
GitHub security best practices are not opposing this approach.
https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-secrets
They, however, are mentioning, that secret should not be structured data, like our Kubeconfig secret is, as there is a risk that the data won't be detected in the terminal output. However, I tested that by adding secret TEST_SECRET with mock Kubeconfig. It was redacted just fine
https://github.com/redhat-openshift-ecosystem/operator-pipelines/runs/4136073792?check_suite_focus=true#step:4:126